worlds/sc2/Client.py

```diff
@@ -1364,8 +1364,8 @@ def mission_reqs_completed(ctx: SC2Context, mission_name: str, missions_complete
                 else:
                     req_success = False
 
-            # Grid-specific logic (to avoid long path checks and infinite recursion)
-            if ctx.mission_order in (MissionOrder.option_grid, MissionOrder.option_mini_grid, MissionOrder.option_medium_grid):
+            # Grid and Blitz-specific logic (to avoid long path checks and infinite recursion) (Includes legacy grids)
+            if ctx.mission_order in (MissionOrder.option_grid, MissionOrder.option_blitz) or (ctx.slot_data_version <= 3 and ctx.mission_order in (3, 4, 8)):
```

Comment:
I'd recommend pulling the magic numbers 3, 4, and 8 out as named constants somewhere. Maybe add them to options.py as constants, like `LEGACY_MEDIUM_GRID = 3`. If they're right next to the option, it reduces the likelihood we forget about the legacy meaning and add new options back using the old numbers (though I suppose the slot data version check should help there as well)

Reply:
Added a legacy grids constant to options.